content/4_Gold/Meet_In_The_Middle.mdx

```diff
@@ -0,0 +1,220 @@
+---
+id: meet-in-the-middle
+title: 'Meet In The Middle'
+author: Chongtian Ma
+description:
+  'Divide the search space into two'
+frequency: 1
+prerequisites:
+---
+
+## Meet In The Middle
+
+<FocusProblem problem="sample" />
+
+### Tutorial
+
+<Resources>
+	<Resource source="CPH" title="5.5 - Meet in the Middle" />
+	<Resource
+		source="Errichto"
+		title="Meet in the Middle | Tutorial & Problems"
+		url="https://www.youtube.com/watch?v=18sJ3mK173s"
+		starred= "true"
+	/>
+</Resources>
+
+<Info title="Pro Tip">
+
+Meet in the middle problems often have double the constraints of an already
```

Comment:
what does this mean?

Reply:
MIM technique can take advantage of the smaller constraint and calculate the solution in two halves. Therefore the constraints tend to be doubled as opposed to a normal problem of this type.

It's hard to explain, do you have any suggestions?

Reply:
IDK, I just found it odd that the 2^N brute force is being referred to as "optimized"

Reply:
it's referring in general, not just this sample problem lol

Reply:
In general, it doesn't really make sense to refer to the simplest possible exponential time solution as optimized. maybe "naive" instead of "optimized"?

Reply:
i think it depends on the problem, because for some more advanced problems you tend to need to optimize a naive solution further and do MIM in order to pass

Reply:
brute force is what I mean, could you give an example of what you mean?

Reply:
for example bitwise dp

you can loop through all subsets or optimize it using dp

in this case i'm talking about the dp optimization with MIM

Reply:
I'm confused, are you saying that "loop through all subsets" is an "already optimized solution"?

Reply:
no, i'm saying that loop through all permutations is a naive solution while optimizing it using bitwise dp is an optimized solution

Reply:
Can you link the problem you are referring to? I thought you were referring to https://judge.yosupo.jp/problem/maximum_independent_set but there's no notion of looping through permutations here.

Reply:
i wasn't referring to a specific MIM problem

I just made up an example of "naive" VS "optimized" solution in my terms

Reply:
ok, can you provide a real example instead of a made-up one? :)

Reply:
https://codeforces.com/problemset/problem/912/E?f0a28=2

From the editorial:
![image](https://user-images.githubusercontent.com/79723780/170548607-0e397a9c-772a-4f63-89af-2051c25db1e1.jpeg)

Brute force cannot pass, you have to optimize the size of the partitions.

Reply:
so meet in the middle is a way to speed up the brute force method of counting the total number of integers <= x, right?

Reply:
yeah i guess, but further optimizations are required to solve the problem.

i see what you mean now.